scripts/partition_manager.py

```diff
@@ -405,26 +415,38 @@ def align_partition(current, reqs, move_up, dynamic_partitions):
                'region': reqs[dynamic_partitions[0]]['region'],
                'placement': {'before' if move_up else 'after': [current]}}
 
-    if current not in dynamic_partitions:
-        # The size of this partition is statically defined. This means the size of the dynamically sized partitions must
-        # be decreased correspondingly. Hence, all partitions which share size with dynamic partition must have their
-        # size reduced. Note that the total amount of 'stealing' is divided between the partitions sharing size with app
-        # (including dynamic partition itself).
-        for p in dynamic_partitions:
-            reqs[p]['size'] = reqs[p]['size'] - (reqs[e]['size'] // len(dynamic_partitions))
+    # Adjust size of dynamic partitions to accommodate for new empty partition
+    for p in dynamic_partitions:
+        reqs[p]['size'] -= reqs[e]['size'] // len(dynamic_partitions)
 
     return e
 
 
-def align_dynamic_partition(app_dep_parts, current, reqs, required_offset):
-    # Since this is a dynamic partition, the introduced empty partition will take space from the dynamic_partition
-    # and the partition being aligned. Take special care to ensure the offset becomes correct.
-    required_offset *= 2
-    for p in app_dep_parts:
-        reqs[p]['size'] -= required_offset // 2
-    reqs[current]['address'] -= required_offset
-    empty_partition_address = reqs[current]['address'] + reqs[current]['size']
-    empty_partition_size = required_offset
+def get_empty_part_to_move_dyn_part(app_dep_parts, current, reqs, num_bytes_to_move_dyn_part, dp, move_end, solution):
+    # The current partition is dynamic, and all partitions between `dp` and `current` must also be dynamic.
+    num_dyn_part_in_front = solution.index(current) - solution.index(dp)
+
+    assert num_bytes_to_move_dyn_part > 0, "Invalid move size specified, must be > 0."
+
+    if move_end:
+        # Add the current partition to the list of dynamic partitions in front,
+        # since the end is moved, and the resize in the current partition will
+        # affect the alignment as well.
+        num_dyn_part_in_front += 1
+
+    reduction_each_dynamic_part = num_bytes_to_move_dyn_part // num_dyn_part_in_front
+
+    assert reduction_each_dynamic_part % 4 == 0, "Reducing partition with non-word aligned size. " + ALIGNMENT_ERROR
+    assert reduction_each_dynamic_part < reqs[dp]['size'], "Size of reduction in size of dynamic partition is bigger " \
+                                                           "than the current size of the dynamic partition. " \
```

Comment:
Does this make it clearer to the user what failed ?
Does the user have any idea what `Size of reduction` mean in this context.

Reply:
It is not easy to tell the user what they should do, because its not easy to know how to fix this issue. It may not be that they are running out of flash, but that the second aligned dynamic partition can not be aligned without running out of flash. And aligning two dynamic partitions can work in some cases, so we can not simply reject that configuration.

The only thing we can do in these cases is to tell the user something along the lines of: "Something is wrong, and we can not fix it you have to set up your entire configuration manually through pm static."

Reply:
When reading the code, I find the `reduction` functionality of internal knowledge to this script.
And because of that, I don't expect the user to have any clue in what `Size of reduction` refers to.
From reader point of view when seeing this error, what is being reduced and why ?
And why is there  a reduction, are they throwing away my code....